p2p/pex/pex_reactor.go

```diff
@@ -637,6 +642,9 @@ func (r *Reactor) dialSeeds() {
 	// do not write error message if there were no seeds specified in config
 	if len(r.seedAddrs) > 0 {
 		r.Switch.Logger.Error("Couldn't connect to any seeds")
+		if retry {
+			time.AfterFunc(seedConnectRetryPeriod, func() { r.dialSeeds(true) })
```

Comment:
I would recommend you to reimplement this recursion call on for loop

Reply:
This will make this code synchronous. My intention was to make the retry asynchronous.

Reply:
Moved the loop to the caller function